Lib/test/test_support.py

```diff
@@ -24,6 +26,73 @@
 TESTFN = os_helper.TESTFN
 
 
+class LogCaptureHandler(logging.StreamHandler):
+    """
+    A logging handler that stores log records and the log text.
+
+    derrived from pytest caplog
+
+    The MIT License (MIT)
+
+    Copyright (c) 2004 Holger Krekel and others
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy of
+    this software and associated documentation files (the "Software"), to deal in
+    the Software without restriction, including without limitation the rights to
+    use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+    of the Software, and to permit persons to whom the Software is furnished to do
+    so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    """
+
+    def __init__(self) -> None:
+        """Create a new log handler."""
+        super().__init__(io.StringIO())
+        self.records = []
+
+    def emit(self, record: logging.LogRecord) -> None:
```

Comment:
Maybe add a true assertion here instead of a type hint?

Reply:
Woops I meant to strip all the type hints here

```suggestion
    def emit(self, record) -> None:
```